From c5966e78612e3cc7834c2ce90713821bc8fdba70 Mon Sep 17 00:00:00 2001
From: Wen Gong <quic_wgong@quicinc.com>
Date: Thu, 2 Nov 2023 19:14:22 -0800
Subject: [PATCH 7/9] SME: Skip wpa_scan_res_match() in wpas_ml_element() for
 MLO connection

It leads SSR with single link.
Neighbor bssid : 00:03:7f:12:cb:cb
wls1: 0: 00:03:7f:12:cb:cb ssid='GONGWEN-WKK-MLO-T-SCAN' wpa_ie_len=0 rsn_ie_len=20 caps=0x511 level=-31 freq=5620
wls1:    skip - BSSID ignored (count=2 limit=0)
MLD: Neighbor doesn't match current SSID - skip link

Signed-off-by: Wen Gong <quic_wgong@quicinc.com>
Signed-off-by: Amutha Ravikumar <quic_aravikum@quicinc.com>
---
 wpa_supplicant/sme.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/wpa_supplicant/sme.c
+++ b/wpa_supplicant/sme.c
@@ -512,9 +512,11 @@ static bool wpas_ml_element(struct wpa_s
 			u8 link_id = *(data + 1) & 0xF;
 
 			if (neigh_bss) {
-				if (wpa_scan_res_match(wpa_s, 0, neigh_bss,
-						       wpa_s->current_ssid,
-						       1, 1)) {
+				wpa_scan_res_match(wpa_s, 0, neigh_bss,
+						   wpa_s->current_ssid,
+						   1, 1);
+
+				if (true) {
 					wpa_s->valid_links |= BIT(link_id);
 					os_memcpy(wpa_s->links[link_id].bssid,
 						  ap_info->data + 1, ETH_ALEN);
